spec.html

```diff
@@ -28546,23 +28359,27 @@ <h1>Math.fround ( _x_ )</h1>
       <emu-clause id="sec-math.hypot">
         <h1>Math.hypot ( _value1_, _value2_, ..._values_ )</h1>
         <p>`Math.hypot` returns an implementation-approximated value representing the result of the square root of the sum of squares of its arguments.</p>
-        <ul>
-          <li>
-            If no arguments are passed, the result is *+0*.
-          </li>
-          <li>
-            If any argument is *+&infin;*, the result is *+&infin;*.
-          </li>
-          <li>
-            If any argument is *-&infin;*, the result is *+&infin;*.
-          </li>
-          <li>
-            If no argument is *+&infin;* or *-&infin;*, and any argument is *NaN*, the result is *NaN*.
-          </li>
-          <li>
-            If all arguments are either *+0* or *-0*, the result is *+0*.
-          </li>
-        </ul>
+        <emu-alg>
+          1. Let _v1_ be ? ToNumber(_value1_).
+          1. Let _v2_ be ? ToNumber(_value2_).
+          1. If _v1_ is *NaN* or _v2_ is *NaN*, return *NaN*.
+          1. If _v1_ is *+&infin;* or _v2_ is *+&infin;*, return *+&infin;*.
+          1. If _v1_ is *-&infin;* or _v2_ is *-&infin;*, return *-&infin;*.
+          1. Let _hasNaN_ be *false*.
+          1. Let _onlyZero_ be *true*.
+          1. If _v1_ is not 0 and _v2_ is not 0, set _onlyZero_ to *false*.
+          1. Repeat, while _values_ is not empty,
+            1. Let _value_ be the first element of _values_.
+            1. Remove the first element from _values_.
+            1. Let _v_ be ? ToNumber(_value_).
+            1. If _v_ is *+&infin;*, return *+&infin;*.
+            1. If _v_ is *-&infin;*, return *-&infin;*.
+            1. If _v_ is *NaN*, set _hasNaN_ to *true*.
+            1. If _v_ is not *+0* or _v_ is not *-0*, set _onlyZero_ to be *false*.
+          1. If _hasNaN_ is *true*, return *NaN*.
+          1. If _onlyZero_ is *true*, return *+0*.
+          1. Return an implementation-approximated value representing the result of the square root of the sum of squares of its arguments.
```

Comment:
I don't think this step is sufficient. I think you should construct a List of all the `ToNumber` results like

```
1. Let _numbers_ be a List containing _v1_ and _v2_.
...
1. Repeat, while _values_ is not empty,
  ...
  1. Append _v_ to _numbers_.
```

and then say here something like

```
1. Return an implementation-approximated value representing the square root of the sum of squares of the elements of _numbers_.
```

Does that make sense?

Reply:
I took a slightly different approach that suggested but let me know if it doesn't work or follow convention.

```
        <emu-alg>
          1. Let _numbers_ be a List containing _value1_ and _value2_, and the elements of _values_ in order.
          1. Let _onlyZero_ be *true*.
          1. Repeat, while _numbers_ is not empty,
            1. Let _number_ be the first element of _numbers_.
            1. Remove the first element from _numbers_.
            1. Let _n_ be ? ToNumber(_number_).
            1. If _n_ is *NaN*, _n_ is *+&infin;*, or _n_ is *-&infin;*, return _n_.
            1. If _n_ is not *+0* or _n_ is not *-0*, set _onlyZero_ to be *false*.
          1. If _onlyZero_ is *true*, return *+0*.
          1. Return an implementation-approximated value representing the result of the square root of the sum of squares of the elements of _numbers_.
        </emu-alg>
```

I used the similar `List` construction sentence proposed below of all the values. Since NaN and +/- infinity all cause a bail out, I thought it was easier to grok by iterating over the original list and returning the exception cases.